allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -452,24 +477,56 @@ def attend_on_question(self,
         # (group_size, encoder_output_dim)
         return util.weighted_sum(encoder_outputs, question_attention_weights)
 
+    def _embed_actions(self, actions: List[List[int]]) -> Tuple[Variable, Variable]:
+        """
+        Returns an embedded representation of the given actions.  The input is assumed to have
+        shape ``(group_size, num_actions)``, though the number of actions for each group element
```

Comment:
Maybe it would be clearer to say something like: "The inputs are a list of valid actions for each element in the group, which may have variable length. These are then padded to have shape (group_size, num_actions) and embedded."

Reply:
:+1:, done.